core-web/libs/portlets/edit-ema/portlet/src/lib/services/inline-edit/inline-edit.service.ts

```diff
@@ -67,23 +67,51 @@ export class InlineEditService {
 
         this.$isInlineEditingEnable.set(true);
 
-        if (!doc.querySelector('script[data-inline="true"]')) {
-            const script = doc.createElement('script');
-            script.dataset.inline = 'true';
-            script.src = '/html/js/tinymce/js/tinymce/tinymce.min.js';
+        if (doc.querySelector('script[data-inline="true"]')) {
+            return;
+        }
+
+        const script = doc.createElement('script');
+        script.dataset.inline = 'true';
+        script.src = '/html/js/tinymce/js/tinymce/tinymce.min.js';
 
-            const style = doc.createElement('style');
-            style.innerHTML = `
+        const style = doc.createElement('style');
+
+        style.innerHTML = `
             [data-inode][data-field-name][data-mode] {
                 cursor: text;
                 border: 1px solid #53c2f9 !important;
                 display: block;
             }
         `;
 
-            doc.body?.appendChild(script);
-            doc.body?.appendChild(style);
-        }
+        // This should be a file in `/html/js/` folder
+        const scriptBlockEditor = doc.createElement('script');
+        scriptBlockEditor.innerHTML = `
+        document.addEventListener('DOMContentLoaded', () => {
+            const editBlockEditorNodes = document.querySelectorAll('[data-block-editor-content]');
+            if (!editBlockEditorNodes.length) {
+                return;
+            }
+            editBlockEditorNodes.forEach((node) => {
+                node.classList.add('dotcms__inline-edit-field');
+                node.addEventListener('click', (event) => {
+                    const dataset = Object.assign({}, node.dataset);
+                     window.parent.postMessage(
+                        {
+                            action: 'editor-inline-editing',
+                            payload: { dataset }
+                        },
+                        '*'
+                    );
+                });
+            });
+        });
+        `;
```

Comment:
Can we just use the block editor Angular component directly?

Reply:
Mmm... Didn't get that question.

This is for traditional pages. We need to track when someone clicks on the Block Editor content and notify the UVE of that action. This approach is similar to how we handle `WYSIWYG` content.

Reply:
I moved this to a separate file; however, all scripts required for traditional pages should be inside the `sdk-editor-vtl.js` file.

Reply:
Traditional or not the block editor loads in the angular side... so it should be a component.

Reply:
I think this whole code, should be on our SDK and make it as reusable as possible for both traditional and headless

Reply:
I already have a customer complaning that the tinmymce needs to be in their building even when they're not using it.

The block editor doesn't load in the customer page it loads in a primeng sidepanel in UVE.

Reply:
We already discussed this!

Reply:
**FYI**: 

1. This was not the block editor but a script to notify the UVE that a content was clicked
2. It's going to be moved to the SDK.vtl - I already have added comments about that.